src/envoy/http/service_control/handler_impl.cc

```diff
@@ -107,9 +121,16 @@ ServiceControlHandlerImpl::ServiceControlHandlerImpl(
 
 ServiceControlHandlerImpl::~ServiceControlHandlerImpl() {}
 
-void ServiceControlHandlerImpl::fillFilterState(FilterState& filter_state) {
+void ServiceControlHandlerImpl::fillFilterState(const RequestHeaderMap& headers,
+                                                FilterState& filter_state) {
   utils::setStringFilterState(filter_state, utils::kFilterStateApiKey,
                               api_key_);
+
+  auto* entry = headers.get(Headers::get().Path);
+  if (entry) {
+    utils::setStringFilterState(filter_state, utils::kFilterStateOriginalPath,
```

Comment:
Would the filter state be updated if the path_matcher filter doesn't find a corresponding operation?

Reply:
You are right. We need put set this filterState before rejection in path matcher filter.